add tsp-lp pool contract

terryyyyyy opened the pull request.

terryyyyyy committed:
add tsp-lp pool contract

tolak commented on contracts/TspLPPooling.sol:
```diff
@@ -0,0 +1,259 @@
+        if (pending > 0)
+            delegators[msg.sender].availablePeanuts = delegators[msg.sender].availablePeanuts.add(pending);
+        
+        require(delegators[msg.sender].availablePeanuts <= Pnuts.balanceOf(address(this)), "ERC20: transfer amount exceeds balance");
```
If we can bump an EVENT and return rather than revert this transaction so that daemon can catch this situation and then trigger TSPPool contract update(for example deposit 1 sp, making reward distribute to Dev address), or if no user trigger TSPPool contract for a long time, we'd hang up here all the time.

tolak commented:
LGTM

terryyyyyy committed:
add event when contract has not enough pnuts

terryyyyyy committed:
fix typo

tolak commented on contracts/TspLPPooling.sol:
```diff
@@ -143,13 +143,19 @@ contract TspLPPooling is Ownable {
+        uint256 balanceOfPnut = Pnuts.balanceOf(address(this));
+
+        if (delegators[msg.sender].availablePeanuts > balanceOfPnut){
+            emit NotEnoughBlanceOfPnut();
```
You haven't declared this event yet. BTW, i prefer to use "InsufficientBalance" here

terryyyyyy committed:
update event name of InsufficientBalance

terryyyyyy committed:
add ignore

terryyyyyy committed:
fix TSPLPToVests calculate

terryyyyyy committed:
delete delegator monitor at deposit method

terryyyyyy committed:
update algorithm

terryyyyyy committed:
delete lastRewardsBlock of PenutsPool

terryyyyyy committed:
when user comein add total TSPLPAmount

terryyyyyy committed:
add getBalanceOfpeanuts method;
return if there is not enough peanuts;
withdraw peanuts when delegator cancel delegation

terryyyyyy committed:
clear available when withdrawpeanuts

terryyyyyy committed:
fix withdraw bug

terryyyyyy committed:
do thing if insufficient peanuts when withdraw peanuts

terryyyyyy committed:
rename contract and updateData method

terryyyyyy committed:
update contract name
